src/components/FilterBuilder/FilterBuilder.ts

```diff
@@ -0,0 +1,182 @@
+import { features } from '../../config';
+
+/**
+ * A fluent builder for constructing structured Algolia-compatible filter strings.
+ *
+ * This class helps compose complex query filters using `AND`, `OR`, negation (`andRaw`),
+ * and conditional logic like `filterByCatalogUuids()` or `excludeVideoContentIfFeatureDisabled()`.
+ *
+ * Useful for dynamic, user-driven search UIs where filters must be assembled programmatically.
+ *
+ * Example:
+ * ```ts
+ * const filter = new FilterBuilder()
+ *   .and('type', 'course')
+ *   .or('level', ['beginner', 'intermediate'])
+ *   .andRaw('NOT content_type:video')
+ *   .filterByCatalogUuids(['c1', 'c2'])
+ *   .build();
+ *
+ * // Result:
+ * // "type:course AND (level:beginner OR level:intermediate) AND NOT
+ * // content_type:video AND (enterprise_catalog_uuids:c1 OR enterprise_catalog_uuids:c2)"
+ * ```
+ *
+ * Available methods:
+ * - `.and(attribute, value)`
+ * - `.or(attribute, values[])`
+ * - `.andRaw(clause)`
+ * - `.filterByCatalogQueryUuids(...)`
+ * - `.filterByCatalogUuids(...)`
+ * - `.filterByEnterpriseCustomerUuid(...)`
+ * - `.excludeVideoContentIfFeatureDisabled()`
+ * - `.build()`
+ */
+export default class FilterBuilder {
+  private filters: string[] = [];
+
+  /**
+   * Adds an AND clause with a single `attribute:value` pair.
+   *
+   * @param attribute - The name of the attribute to filter on.
+   * @param value - The value the attribute must match.
+   * @returns The current FilterBuilder instance for chaining.
+   *
+   * @example
+   *   new FilterBuilder().and('type', 'course').build()
+   *   // → "type:course"
+   */
+  and(attribute: string, value: string): this {
```

Comment:
nit: TS should be able to infer the return type of this method as `this` without an explicit type hint. Same for the other class methods within `FilterBuilder`.